borg/fuse.py

```diff
@@ -76,6 +76,16 @@ def process_archive(self, archive, prefix=[]):
             data = self.key.decrypt(key, chunk)
             unpacker.feed(data)
             for item in unpacker:
+                try:
+                    # This can happen when an archive was created with a command line like
+                    # $ borg create ... dir1/file dir1
+                    # In this case the code below will have created a default_dir inode for dir1 already.
+                    inode = self._find_inode(item[b'path'], prefix)
+                except KeyError:
+                    pass
+                else:
+                    self.items[inode] = item
```

Comment:
looks good. i first thought we have to put it into that cache, but that applies only to files.